src/js/hooks/cycleCount/useResolveStepTable.jsx

```diff
@@ -270,6 +259,37 @@ const useResolveStepTable = ({
         if (rowIndex !== null && columnId && error !== null) {
           setError(null);
         }
+        const handleClick = (event) => {
+          if (rowIndex === null && columnId === null) {
+            return;
+          }
+
+          const { target } = event;
+          const isInputElement = target.closest('input, select, textarea, .date-field-input, .react-datepicker, .react-select__control');
+          const isDatePickerDay = target.closest('.react-datepicker__day');
+          const isSelectOption = target.closest('.react-select__option');
+
+          // if this is input element, then we don't want to reset rowIndex and columnId,
+          // and re-render the component again because then all tables will be re-rendered
+          // which will cause performance issues
+          if (!isInputElement && !isDatePickerDay && !isSelectOption) {
+            setRowIndex(null);
+            setColumnId(null);
+            forceRerender();
+          }
+
+          // if this is date picker day or select option, then we want to reset rowIndex
+          // and columnId because then we close the date picker or select dropdown
+          if (isDatePickerDay || isSelectOption) {
+            setRowIndex(null);
+            setColumnId(null);
+          }
+        };
+        document.addEventListener('click', handleClick);
+
+        return () => {
+          document.removeEventListener('click', handleClick);
+        };
```

Comment:
the same as in the previous hook - add more comments explaining how it really solves the performance issue.